extensions/common/azure/azure-blob-core/src/main/java/org/eclipse/edc/azure/blob/adapter/DefaultBlobAdapter.java

```diff
@@ -24,15 +28,33 @@
  * Implementation of {@link BlobAdapter} using a {@link BlockBlobClient}.
  */
 public class DefaultBlobAdapter implements BlobAdapter {
+    private final long blockSizeInMb;
+    private final int maxConcurrency;
+    private final long maxSingleUploadSizeInMb;
     private final BlockBlobClient client;
 
-    public DefaultBlobAdapter(BlockBlobClient client) {
+    public DefaultBlobAdapter(
+            BlockBlobClient client, Long blockSizeInMb, Integer maxConcurrency, Long maxSingleUploadSizeInMb) {
+        this.blockSizeInMb = blockSizeInMb;
+        this.maxConcurrency = maxConcurrency;
+        this.maxSingleUploadSizeInMb = maxSingleUploadSizeInMb;
         this.client = client;
     }
 
     @Override
     public OutputStream getOutputStream() {
-        return client.getBlobOutputStream(/* overwrite = */ true);
+        return getOutputStream(null);
+    }
+
+    @Override
+    public OutputStream getOutputStream(ProgressListener progressListener) {
```

Comment:
progressListener looks to be an additional feature that is not in scope with this PR and it's not tested properly too.
Please just leave a single `getOutputStream` method without using the ProgressListener and eventually open another issue for that feature.

Reply:
I thought it could be included here, but you're are right. Updated in [here](https://github.com/eclipse-edc/Technology-Azure/pull/298/commits/559b3dfd11a416d93c004ac0af10650f69fa1486), thank you!